yt_dlp/extractor/vk.py

```diff
@@ -710,21 +672,20 @@ def _real_extract(self, url):
 
         for audio in re.findall(r'data-audio="([^"]+)', webpage):
             audio = self._parse_json(unescapeHTML(audio), post_id)
-            a = self._AUDIO._make(audio[:16])
-            if not a.url:
+            if not audio['url']:
                 continue
-            title = unescapeHTML(a.title)
-            performer = unescapeHTML(a.performer)
+            title = unescapeHTML(audio['title'])
+            artist = unescapeHTML(audio['artist'])
             entries.append({
-                'id': '%s_%s' % (a.owner_id, a.id),
-                'url': self._unmask_url(a.url, a.ads['vk_id']),
-                'title': '%s - %s' % (performer, title) if performer else title,
-                'thumbnails': [{'url': c_url} for c_url in a.cover_url.split(',')] if a.cover_url else None,
-                'duration': int_or_none(a.duration),
+                'id': '%s_%s' % (audio['owner_id'], audio['id']),
+                'url': audio['url'],
```

Comment:
I think I found the issue. Try removing `url`, `ext`, `protocol` and add `'formats': self._extract_m3u8_formats(...)` instead

Reply:
I thought so too, because it works for vk videos, but the files this outputs are no different to the ones before. It just doesn't seem to find anything:
```
[info] Available formats for 2000326341_456244816:
ID EXT RESOLUTION │ PROTO │ VCODEC  ACODEC
───────────────────────────────────────────
0  mov unknown    │ m3u8  │ unknown unknown
```
 I tried it with live=true/false and m3u8_id=None/'hls'. Ffprobe identifies the resulting files as:
```
mpegts @ 0x5614dee86000] start time for stream 1 is not set in estimate_timings_from_pts
Input #0, mpegts, from 'input.mp4':
  Duration: 00:06:10.44, start: 0.127122, bitrate: 352 kb/s
  Program 1 
  Stream #0:0[0x100]: Audio: mp3 ([3][0][0][0] / 0x0003), 44100 Hz, stereo, fltp, 320 kb/s
  Stream #0:1[0x101]: Data: timed_id3 (ID3  / 0x20334449)
Unsupported codec with id 98313 for input stream 1
```
This will not be playable with  VLC, which would be my standard for "working" media files. MPV however manages to play that. If you fix them with `ffmpeg -i input.mp4  -map 0 -dn -ignore_unknown -c copy -f mp4 output.mp4` they become:
```
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'output.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2mp41
    encoder         : Lavf59.27.100
  Duration: 00:06:10.44, start: 0.000000, bitrate: 321 kb/s
  Stream #0:0[0x1](und): Audio: mp3 (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 320 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
      vendor_id       : [0][0][0][0]
```
These are now playable without any issues everywhere. Given that m3u8 already has a fixup for broken audio (that just doesnt work here because of mp3 instead of aac audio) I would guess, that this is a somewhat known problem, that a user is not expected to solve on their own?

Reply:
Try inserting `{'vcodec': 'none', 'acodec': 'mp3', 'container': 'm4a_dash'}` into the format